File: Elearn-DataTier/src/main/java/via/dk/elearn/models/Lecture.java
package via.dk.elearn.models;

import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

import javax.persistence.*;
import java.time.LocalDate;

@Entity
@Data
@Builder
@AllArgsConstructor
@NoArgsConstructor
public class Lecture {
    @Id
    @GeneratedValue(strategy = GenerationType.AUTO)
    private Long id;

    @Column(name = "title", length = 50)
    private String title;

    @Column(name = "description", length = 80)
    private String description;

    @Column(unique = true, name = "url")
    private String url;

    @Lob
    @Column(name = "body")
    private String body;

    @Lob
    @Column(name = "image")
    private String image;

    @Column
    private LocalDate date;

    @ManyToOne(fetch = FetchType.EAGER)
    @JoinColumn(name = "course_id")
    private Course course;

    @ManyToOne(fetch = FetchType.EAGER)
    @JoinColumn(name = "teacher_id")
    private Teacher teacher;

    public Lecture(String title, String url, String image, String body) {
        this.title = title;
        this.url = url;
        this.image = image;
        this.body = body;
    }

    public Lecture(String title, String description, String url, String body, String image, LocalDate date, Course course, Teacher teacher) {
        this.title = title;
        this.description = description;
        this.url = url;
        this.body = body;
        this.image = image;
        this.date = date;
        this.course = course;
        this.teacher = teacher;
    }
}

File: Elearn-ApplicationTier/Elearn.Shared/Models/Lecture.cs
using System.ComponentModel.DataAnnotations;

namespace Elearn.Shared.Models;
public class Lecture
{
    public long Id { get; set; }

    [Required, StringLength(20, ErrorMessage = "Please use only 20 characters")]
    public string Url { get; set; }
    public string Image { get; set; }
    
    [Required] public string Title { get; set; }
    
    public string Body { get; set; }
    
    public string Description { get; set; }
    
    public DateTime DateCreated { get; set; } = DateTime.Now;
    public Teacher Author { get; set; }

    public Lecture(string url, string image, string title, string body, string description, Teacher author)
    {
        Url = url;
        Image = image;
        Title = title;
        Body = body;
        Description = description;
        Author = author;
    }

    public Lecture(long id, string url, string image, string title, string body, string description, Teacher author)
    {
        Id = id;
        Url = url;
        Image = image;
        Title = title;
        Body = body;
        Description = description;
        Author = author;
    }


    public Lecture()
    {
        
    }
}